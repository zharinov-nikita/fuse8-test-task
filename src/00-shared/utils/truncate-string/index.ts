export * from './truncate-string.util'
