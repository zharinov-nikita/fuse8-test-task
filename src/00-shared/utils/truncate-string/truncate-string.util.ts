export function sharedUtilTruncateString(
    string: string,
    maxLength: number,
    textEnd?: string
) {
    if (string.length > maxLength) {
        return `${string.slice(0, maxLength)}${textEnd ? textEnd : '...'}`
    } else {
        return string
    }
}
