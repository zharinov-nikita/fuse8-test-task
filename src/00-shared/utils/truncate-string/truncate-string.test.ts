import { describe, test, expect } from 'vitest'
import { sharedUtilTruncateString } from './truncate-string.util'

describe('sharedUtilTruncateString', () => {
    test('case 1', () => {
        const string = 'hello world'
        const maxLength = 3
        const result = 'hel...'
        expect(sharedUtilTruncateString(string, maxLength)).toEqual(result)
        expect(sharedUtilTruncateString(string, maxLength)).toMatchSnapshot()
    })
    test('case 2', () => {
        const string = 'hello world'
        const maxLength = 5
        const result = 'hello world'
        expect(sharedUtilTruncateString(string, maxLength, ' world')).toEqual(
            result
        )
        expect(sharedUtilTruncateString(string, maxLength, ' world'))
            .toMatchSnapshot
    })
})
