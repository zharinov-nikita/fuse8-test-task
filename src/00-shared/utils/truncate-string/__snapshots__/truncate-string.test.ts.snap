// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sharedUtilTruncateString > case 1 1`] = `"hel..."`;

exports[`sharedUtilTruncateString > snapshot 1`] = `"hello world"`;
